__tests__/Titled.test.tsx

```diff
@@ -41,3 +55,23 @@ test('should renrender nested title', () => {
   expect(handleChange).toHaveBeenCalledTimes(2);
   expect(handleChange.mock.calls[1][0]).toBe('A');
 });
+
+test('should handle string literal title', () => {
+  const handleChange = vi.fn();
+  render(<Titled title={'A'} onChange={handleChange}></Titled>);
+  expect(handleChange).toHaveBeenCalledTimes(1);
+  expect(handleChange.mock.calls[0][0]).toBe('A');
+});
+
+test('should handle nested string literal title', () => {
+  const handleChange = vi.fn();
+  render(
+    <Titled title={() => 'A'} onChange={handleChange}>
+      <Titled title={'B'}>
+        <Titled title={title => `C | ${title}`}>Child</Titled>
+      </Titled>
+    </Titled>
+  );
+  expect(handleChange).toHaveBeenCalledTimes(1);
+  expect(handleChange.mock.calls[0][0]).toBe('C | B');
```

Comment:
This is a little surprising.  I'd expect an `onChange` for the inner-most component to be `C | B`, but why is the onChange for the outer parent that as well, and not `A`?

Reply:
The change is controlled by the inner-most component but it is _dispatched_ by the outermost component. Very neat pattern.